### ISSUE TITLE
Moves @statements outside the CSS rule set

### ISSUE THREAD

**Issue by chaucerbao** [postcss/postcss-nested#11]
Input:
```css
.article {
  .title {
    @statement arguments;
    color: blue;
  }
}
```

Expected:
```css
.article .title {
  @statement arguments; /* Should stay inside ".article .title" rule set */
  color: blue;
}
```

Actual:
```css
.article .title {
  color: blue;
}
@statement arguments; /* Appears outside */
```

I see how this may be useful for @media queries, but for other PostCSS plugins, such as [postcss-simple-extend](https://github.com/davidtheclark/postcss-simple-extend) where the context is important, the @statement should stay inside.

**Comment by ai** [comment]
We can unwrap only whitelist of at-rule:

- `@media`
- `@supports`
- `@document`

Does I remember all at-rules?

**Comment by chaucerbao** [comment]
I surely don't use all of these, but it seems to be a good starting reference:
https://developer.mozilla.org/en-US/docs/Web/CSS/At-rule

Nested at-rules
"... can be used as a statement of a style sheet as well as inside of conditional group rules"

**Comment by ai** [comment]
Because this fix will broke API, I made it after PostCSS 4.1 release to combine broking changes.

**Comment by ai** [comment]
Also I add a option to set custom at-rules whitelist.

**Comment by chaucerbao** [comment]
Sounds good, can't wait to see it!

**Comment by timkelty** [comment]
PostcSS 4.1 has been released, have these changes been made?

**Comment by ai** [comment]
I will wrote a PostCSS plugins guidelines on this weekend and them update all my plugins to it. So I think issue will be fixed before 3rd of May.

**Comment by ai** [comment]
Sorry, for waiting. I spend all my free working hours in this week on Helsinki talk :).

**Comment by ai** [comment]
Done 5c0f881

**Comment by ai** [comment]
Released in 0.3 version
